optimade/grammar/v0.10.0.lark

```diff
@@ -0,0 +1,90 @@
+// optimade v0.10.0 grammar spec in lark grammar format
+
+?start: filter
+filter: expression*
```

Comment:
I'll just reiterate my previous questions here concerning the still open conversation from a previous review:
> I just realised that probably I partly misunderstood the question. So `filter` should be just a rule (there is no matching to the "filter word) and the `?start` just means that the `start` rule will be skipped from the `Tree`. Based on the current implementation of the `Transformer`, it looks like that we can use the same '?' trick for the `value`, `constant`, `comparison` rules as well to simplify the Lark Tree.

How far are you in this "optimization"? Do you need some assistance? Will be happy to try and help and push some changes - although it will change the transformers slightly.

Reply:
Although there are few rules that can be removed/suppressed from the parsed tree, I think they are not very significant. You can find these rules in the implementation of the  `Transformer` with a comment: `# Note: Do nothing!`
I didn't remove them yet, because they might be useful for implementing other `variant`s.

Reply:
Thank you! :) That is completely reasonable.